### DISCUSSION TITLE
Default Component for Anchor and Image

### DISCUSSION THREAD

**Discussion by ghost** [mantinedev/mantine#6560] [Q&A]
I love that we are able to alter the `Image` and `Anchor` however it can feel a bit tedious writing `component={Link}` every single time in a Next.js app, and the same for `Image` it would be nice if in `theme.ts` we could define a `componentDefaults` object like so:

``` theme.ts
import NextLink from 'next/link'
import NextImage from 'next/image'

export const theme = createTheme({
  componentDefaults: {
    anchor: NextLink,
    image: NextImage
  }
  // ...rest of configuration
})
```

**Comment by rtivital** [comment]
https://mantine.dev/theming/default-props/

**Reply by ghost** [reply]
Whoops. My bad, I did read through the docs, I must have just missed some bits, there is a lot to take in.

Apologies!

**Reply by ghost** [reply]
@rtivital 

``` typescript
  components: {
    Anchor: Anchor.extend({
      defaultProps: {
        component: NextLink,
      },
    }),
    Image: Anchor.extend({
      defaultProps: {
        component: NextImage,
      },
    }),
  }
  ```

Getting this error `Object literal may only specify known properties, and 'component' does not exist in type 'Partial<AnchorProps> & DataAttributes'.`
